src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts

```diff
@@ -233,6 +236,7 @@ export class ObjectExplorerService implements IObjectExplorerService {
 			if (!session) {
 				return;
 			}
+			this._treeNodeCache.delete(session.sessionId);
```

Comment:
this is the only clean up I've seen so far, I think we should also do clean up for disconnect and refresh operations

Reply:
Yeah, this is the only place where we cleanup session. 
1. In case of disconnect we call this function indirectly. 

2. In case of refresh, I just update the nodes within the cache
 https://github.com/microsoft/azuredatastudio/blob/25974b26e35e1f14c315f8c66f25b8bd32b787f0/src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts#L743-L750

Reply:
in case of refresh, if some nodes are gone (e.g. a table is deleted), the current implementation won't remove them.

Reply:
Maybe when you get a refresh request you look at the tree structure and get all its children and remove those before updating the cache? Should hopefully save having to duplicate the tree structure in the cache itself.

Reply:
Added code for it. In case of a node being refreshed, I am now clearing the entire subtree with the node as parent from the cache now.